src/core/public/utils/mount_point.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect, useRef } from 'react';
+import ReactDOM from 'react-dom';
+import { I18nProvider } from '@kbn/i18n/react';
+
+/**
+ * A function that will mount the banner inside the provided element.
+ * @param element the container element to render into
+ * @returns a {@link UnmountCallback} that unmount the element on call.
+ *
+ * @public
+ */
+export type MountPoint = (element: HTMLElement) => UnmountCallback;
+
+/**
+ * A function that will unmount the element previously mounted by
+ * the associated {@link MountPoint}
+ *
+ * @public
+ */
+export type UnmountCallback = () => void;
+
+/**
+ * MountWrapper is a react component to mount a {@link MountPoint} inside a react tree.
+ *
+ * @internal
+ */
+export const MountWrapper: React.FunctionComponent<{ mount: MountPoint }> = ({ mount }) => {
```

Comment:
I don't think we should expose these at all from core since they are React-based. Is there any reason why `mountReact` and `reactMount` can't be the same implementation in the `kibana-react` package?

Reply:
We are not exporting `mountReact` from core. It's only used internally for tests. I did not want to import `kibana_react` in core for tests, so I had to 'duplicate' here (the different naming was only here to avoid falsy import from IDE). Should I move the core version `mountReact` to a `test_helpers` folder or something similar ?

Reply:
I'm fine with this as long as it's not exported from core. If it is only used in tests, a convention for test helpers would be good. Let's just make sure it's not included in production builds by modifying the `src/dev/build/tasks/copy_source_task` module to exclude directories with that name.

Reply:
There seems to already be a naming convention around `test_helpers` file or directories. I'm guessing they should already be excluded then ?